File: app/src/main/java/com/ireddragonicy/konabessnext/KonaBessApplication.java
package com.ireddragonicy.konabessnext;

import android.app.Application;
import android.content.Context;

import com.ireddragonicy.konabessnext.utils.LocaleUtil;
import com.ireddragonicy.konabessnext.utils.RootHelper;
import com.ireddragonicy.konabessnext.ui.SettingsActivity;
import com.topjohnwu.superuser.Shell;

public class KonaBessApplication extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(LocaleUtil.wrap(base));
    }

    @Override
    public void onCreate() {
        super.onCreate();
        SettingsActivity.applyThemeFromSettings(this);

        // Ensure RootHelper static initializer runs, then pre-cache shell
        try {
            Class.forName(RootHelper.class.getName());
            Shell.getShell(shell -> {
            });
        } catch (Exception ignored) {
        }
    }
}

File: app/src/main/java/com/ireddragonicy/konabessnext/KonaBessApplication.kt
package com.ireddragonicy.konabessnext

import android.app.Application
import android.content.Context
import com.ireddragonicy.konabessnext.utils.LocaleUtil
import com.ireddragonicy.konabessnext.utils.RootHelper
import com.ireddragonicy.konabessnext.ui.SettingsActivity
import com.topjohnwu.superuser.Shell
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class KonaBessApplication : Application() {

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(LocaleUtil.wrap(base))
    }

    override fun onCreate() {
        super.onCreate()
        SettingsActivity.applyThemeFromSettings(this)

        // Ensure RootHelper static initializer runs (or object is initialized), then pre-cache shell
        try {
            // Configure Shell builder before creating the shell
            Shell.enableVerboseLogging = false
            Shell.setDefaultBuilder(Shell.Builder.create()
                .setFlags(Shell.FLAG_REDIRECT_STDERR)
                .setTimeout(10))
            
            // Accessing the object instance to force initialization if needed
            Shell.getShell { }
        } catch (ignored: Exception) {
        }
    }
}
